example_code/assertj/src/test/java/testng/TestNgAssertions.java

```diff
@@ -0,0 +1,130 @@
+package testng;
+
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+public class TestNgAssertions {
+
+    private final Object notNull = new Object();
+    private final Object nullObject = null;
+    private final Object a = new Object();
+    private final Object actualA = a;
+    private final Object b = new Object();
+    private final Object[] actual = List.of().toArray();
+    private final Object[] expected = List.of().toArray();
+
+    @Test
+    void ng_assertArrayEquals() {
+        // Should change to: assertThat(actual).isEqualTo(expected);
+        org.testng.Assert.assertEquals(actual, expected);
+
+        // Should change to: assertThat(actual).as("message").isEqualTo(expected);
+        org.testng.Assert.assertEquals(actual, expected, "message");
+    }
+
+    @Test
+    void ng_assertEquals() {
+        // Should change to: assertThat(3).isEqualTo(3);
+        org.testng.Assert.assertEquals(3, 3);
+
+        // Should change to: assertThat(3).as("message").isEqualTo(3);
+        org.testng.Assert.assertEquals(3, 3, "message");
```

Comment:
minor: same comment as the other file just some way to differentiate the arguments. feel free to ignore if you disagree. Not sure if using an expression is appropriate but it does the trick.
```suggestion
        // Should change to: assertThat(1+2).isEqualTo(3);
        org.testng.Assert.assertEquals(1+2, 3);

        // Should change to: assertThat(1+2).as("message").isEqualTo(3);
        org.testng.Assert.assertEquals(1+2, 3, "message");
```

Reply:
Agreed, I just came back here because I realised I had wanted to mention it. Should help with recipe maintenance.

Reply:
100% agree. Done!